src/transformers/modeling_bert.py

```diff
@@ -597,7 +597,6 @@ class BertPreTrainedModel(PreTrainedModel):
     config_class = BertConfig
     load_tf_weights = load_tf_weights_in_bert
     base_model_prefix = "bert"
-    authorized_missing_keys = [r"position_ids"]
```

Comment:
I don't understand that part: I see that one of the subclass as another `authorized_missing_keys` but you can still have the superclass have this default and only change it for the one subclass that needs a different value.

Reply:
Hmm, yeah I don't like to inherit from a parent class if the value is different => So I think it's better to remove it from the parent class and add it individually for each class because there are two subclasses that are different and I think inheriting an attribute that is different and then overwriting it is not how inheritance should work. 

But really not a big deal to me, so I can revert this so that it is more consistent with the other classes. Would you prefer this @sgugger ?

Reply:
A class attribute is specific to a class. It's bad to change it on different instances of one class (that's what instance attributes are for) but completely ok to have two different subclasses have two different attributes. It's exactly like overriding a method when subclassing.

Reply:
_out-of-scope:_
I'm not a big fan of overriding a method when subclassing either to be honest. *E.g.* I think it's good practice to have a 
```python 
def parent_class_method(...):
      raise NotImplementedError()
```
so that the sub-classes are forced to implement the method (assuming that the sub-classes have at least two sub-classes which implement this function differently, 
**but** I think it's bad practice to have a 
```python
def parent_class_method(...):
     ### logic that applies for 90% of subclasses
    ...
```
and then overwrite this method only for the subclasses which needs a different behavior. I think this case should be avoided because it is not very readable and can lead to problems if people don't see that the method was overwritten.
I would for example force in `generation_utils` to have `prepare_logits_for_generation()` to have a `NotImplementedError()` abstract method, which we currently don't have...

Just as a side-not for my inheritance opinions :D What do you think?

=> Now taking this to the extreme, I would have not overwritten the `missing_keys` attribute either. But in the case of this PR, I am 100% fine with keeping the old logic for consistency and because it is really not hard / important to understand where this class attribute is set for the user. So will revert this change to how it was before.

Reply:
I wouldn't say it's bad practice since it's what every OO course teaches you to do ;-) I guess it depends on the goal and the readability you aim to. I would say your approach is better for things we want to expose to users and think modifications/customization are going to be often done (a bit like our approach to the modeling files) but for internal things, it's okay to use the traditional OO approach.

Anyhow, do what you want for those class attributes are you are the ones actually writing this PR, it was just my two cents ;-)

Reply:
reverted it as suggested by you :-)